frontend_tests/node_tests/router.js

```diff
@@ -0,0 +1,55 @@
+var Router = require("js/router.js")({ addEventListener: function () {} });
+var router = new Router();
+
+// test the parsing components to make sure that they are structurally what we
+// expect them to be.
+assert.deepEqual(router.parse("organization/emoji-settings"), [
+    { type: router.type.PATH, component: "organization" },
+    { type: router.type.PATH, component: "emoji-settings" },
+]);
+
+assert.deepEqual(router.parse("organization/:organization"), [
+    { type: router.type.PATH, component: "organization" },
+    { type: router.type.VARIABLE, component: ":organization" },
+]);
+
+assert.deepEqual(router.parse("narrow/:action/then/:verb"), [
+    { type: router.type.PATH, component: "narrow" },
+    { type: router.type.VARIABLE, component: ":action" },
+    { type: router.type.PATH, component: "then" },
+    { type: router.type.VARIABLE, component: ":verb" },
+]);
+
+assert.deepEqual(router.parse("/narrow/*"), [
+    { type: router.type.PATH, component: "narrow" },
+    { type: router.type.ALL, component: "*" },
+]);
+
+// assert that the path matches a defined generic path.
+assert.equal(!!router.matches("/narrow/:stream", "narrow/Verona"), true);
```

Comment:
`assert(foo)` is better than `assert.equal(!!foo, true`.  It's just as strong an assertion, but more concise.
 Likewise for `assert(!foo)` vs. `assert.equal(!!foo, false)`.

If your `matches` functions return `true` and `false` specifically, that's ideal.